tests/test_interface.rb

```diff
@@ -165,6 +170,37 @@ def interface_supports_property?(intf, message)
     flunk(message)
   end
 
+  # Helper to find first valid speed that isn't "auto"
+  def valid_speed_set(interface)
+    valid_speed = nil
+    speeds = capable_speed_values(interface)
+    speeds.each do |value|
+      next if value == 'auto'
+      begin
+        interface.speed = value
+        assert_equal(value.to_i, interface.speed)
+      rescue Cisco::CliError => e
+        next if speed_change_disallowed?(e.message)
+        raise
+      end
+      # Exit loop once proper speed is found
+      valid_speed = value
+      break
+    end
+    valid_speed
+  end
+
+  # Helper to check for misc speed change disallowed error messages.
+  def speed_change_disallowed?(message)
+    patterns = ['port doesn t support this speed',
+                'Changing interface speed is not permitted',
+                'requested config change not allowed',
+                /does not match the (transceiver speed|port capability)/,
```

Comment:
I see. I got that they're all used as regex patterns but I didn't realize you need the slashes in this case to avoid a literal match of the `(|)` chars.